### PR TITLE
docs: Clarify patch vs major/minor release procedure

### PR DESCRIPTION
**Pull request by dtrudg**

### CODE REVIEW

File: RELEASE_PROCEDURE.md [hunk 1]
```diff
@@ -1,15 +1,79 @@
...
+### Creating the Patch Release
+
+1. Check the version of Go installed in the CI environment from
+   `.circleci/config.yml`. Ensure that it is the latest release, as it will be
+   used to build our release packages. Update if necessary via a PR against the
+   release branch.
+1. Modify the `README.md`, `INSTALL.md`, `CHANGELOG.md` via PR against the
+   release branch, so that they reflect the version to be released.
+1. After all PRs are merged, apply an annotated tag to the HEAD of the release
+   branch via `git tag -a -m "SingularityCE v3.11.2" v3.11.2`.
```
Comment (tri-adam) [inline RELEASE_PROCEDURE.md:47]: Might be worth signing the release tags?
```suggestion
   branch via `git tag -a -s -m "SingularityCE v3.11.2" v3.11.2`.
```

File: RELEASE_PROCEDURE.md [hunk 2]
```diff
@@ -24,68 +88,76 @@ will be working in an up-to-date local clone of the GitHub repository, where the
...
 1. Apply an annotated tag via
-   `git tag -a -m "SingularityCE v3.8.0 Release Candidate 1" v3.8.0-rc.1`.
-1. Push the tag via `git push upstream v3.8.0-rc.1`.
-1. Create a tarball via `mconfig -v && make dist`.
-1. Test intallation from the tarball.
-1. Compute the sha256sum of the tarball e.g. `sha256sum *.tar.gz > sha256sums`.
-1. Create a GitHub release, marked as a 'pre-release', incorporating
-   `CHANGELOG.md` information, and attaching the tarball and `sha256sums`.
-1. Notify the community about the RC via the Google Group and Slack.
+   `git tag -a -m "SingularityCE v3.11.0 Release Candidate 1" v3.11.0-rc.1`.
```
Comment (tri-adam) [inline RELEASE_PROCEDURE.md:112]: Same here, worth signing?

```suggestion
   `git tag -a -s -m "SingularityCE v3.11.0 Release Candidate 1" v3.11.0-rc.1`.
```

File: RELEASE_PROCEDURE.md [hunk 3]
```diff
@@ -1,15 +1,79 @@
...
+1. Check the version of Go installed in the CI environment from
+   `.circleci/config.yml`. Ensure that it is the latest release, as it will be
+   used to build our release packages. Update if necessary via a PR against the
+   release branch.
+1. Modify the `README.md`, `INSTALL.md`, `CHANGELOG.md` via PR against the
+   release branch, so that they reflect the version to be released.
+1. After all PRs are merged, apply an annotated tag to the HEAD of the release
+   branch via `git tag -a -m "SingularityCE v3.11.2" v3.11.2`.
+1. Push the tag to the GitHub repository via `git push upstream v3.11.2`.
+1. (Re-)trigger the CI workflow against the tag.
```
Comment (tri-adam) [inline RELEASE_PROCEDURE.md:49]: Might be worth setting up an explicit CI workflow to automate this ([example](https://github.com/sylabs/sif/blob/8bf9627e2b6c94dfd91e4b1f898fa29b0f173539/.circleci/config.yml#L142-L150))?

File: RELEASE_PROCEDURE.md [hunk 4]
```diff
@@ -24,68 +88,76 @@ will be working in an up-to-date local clone of the GitHub repository, where the
...
    release branch, so that they reflect the version to be released.
-1. Apply an annotated tag via `git tag -a -m "SingularityCE v3.8.0" v3.8.0`.
-1. Push the tag via `git push upstream v3.8.0`.
-1. Create a tarball via `mconfig -v && make dist`.
-1. Test intallation from the tarball.
-1. Compute the sha256sum of the tarball e.g. `sha256sum *.tar.gz > sha256sums`.
-1. Create a GitHub release, incorporating `CHANGELOG.md` information, and
-   attaching the tarball and `sha256sums`.
-1. Notify the community about the RC via the Google Group and Slack.
+1. Apply an annotated tag via `git tag -a -m "SingularityCE v3.12.0" v3.12.0`.
```
Comment (tri-adam) [inline RELEASE_PROCEDURE.md:142]: Same here, worth signing?

```suggestion
1. Apply an annotated tag via `git tag -a -s -m "SingularityCE v3.12.0" v3.12.0`.
```

**Review by tri-adam — COMMENTED** [review]
Nice update, much clearer. Just a few small suggestions, thanks!

File: RELEASE_PROCEDURE.md (hunk 1 above)
Comment (dtrudg) [reply RELEASE_PROCEDURE.md:47]: Holding on this pending possible strategy for cross-siging / publishing keys.

File: RELEASE_PROCEDURE.md (hunk 2 above)
Comment (dtrudg) [reply RELEASE_PROCEDURE.md:112]: Holding on this pending possible strategy for cross-siging / publishing keys.

File: RELEASE_PROCEDURE.md (hunk 4 above)
Comment (dtrudg) [reply RELEASE_PROCEDURE.md:142]: Holding on this pending possible strategy for cross-siging / publishing keys.

File: RELEASE_PROCEDURE.md (hunk 3 above)
Comment (dtrudg) [reply RELEASE_PROCEDURE.md]: [393bb54](https://github.com/sylabs/singularity/pull/1573/commits/393bb54f061f0450771836ab8a52e04fa5430d15) should be enough I believe?

**Pull request merged**